x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStopDatafeedAction.java

```diff
@@ -186,10 +187,14 @@ public void onResponse(PersistentTasksCustomMetaData.PersistentTask<?> persisten
 
                     @Override
                     public void onFailure(Exception e) {
-                        final int slot = counter.incrementAndGet();
-                        failures.set(slot - 1, e);
-                        if (slot == startedDatafeeds.size()) {
-                            sendResponseOrFailure(request.getDatafeedId(), listener, failures);
+                        if ((e instanceof ResourceNotFoundException &&
+                            Strings.isAllOrWildcard(new String[]{request.getDatafeedId()})) == false) {
+
+                            final int slot = counter.incrementAndGet();
+                            failures.set(slot - 1, e);
```

Comment:
Same problem as for jobs - only `failures.set(slot - 1, e);` should be within the scope of the newly added `if`.